### PR TITLE
Adding subtract self to the neighborlist generator

### PR DESCRIPTION
**Pull request by rosecers**

### CODE REVIEW

**Comment by rosecers** [comment]
This PR may rely on #20 before it is clean to merge

**Comment by rosecers** [comment]
@arthur-lin1027 or @acrowley5 can you take a look?

**Review by YCC-ProjBackups — APPROVED** [review]
@rosecers

I looked through this code, and ran my own test script, and it seems to be working fine. I also checked multiple times whether putting `not` in front of the `self.subtract_center_contribution` is necessary, and I am quite confident that putting `not` in front leads to the correct and intended implementation.

The parenthesis around `not self.subtract_center_contribution` is not necessary, but since it will have literally zero impact, I decided to leave it alone.

I will mark this PR as "Approve" for now, but let me know if you want me to do more in-depth analysis (though you may have to be specific, as I don't know what else I need to do)

**Pull request merged**
